spec/jobs/assignment_deadline_submission_job_spec.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe AssignmentDeadlineSubmissionJob, type: :job do
+  before do
+    @primary_mentor = FactoryBot.create(:user)
+    @group = FactoryBot.create(:group, primary_mentor: @primary_mentor)
+    @assignment = FactoryBot.create(:assignment, group: @group)
+  end
+
+  describe "#perform" do
+    it "when the assignment is nil" do
+      expect(described_class.perform_now(nil)).to be_nil
+    end
+
+    it "when the assignment is closed" do
+      @assignment.status = "closed"
+      expect(described_class.perform_now(@assignment.id)).to be_nil
+    end
+
+    describe "when the assignment is open" do
+      it "if deadline has not passed, dont close" do
```

Comment:
```suggestion
      it "if deadline has not passed, don't close" do
```

Just a little type here